### PR TITLE
cf-vault: init at 0.0.11

### PR DESCRIPTION
**Pull request by viraptor**
###### Motivation for this change

Add the new cf-vault package for cloudflare credentials management.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by viraptor** [comment]
Removed the vendoring sha. For some reason ofborg got different result than me, which means I probably shouldn't include it at all.

File: pkgs/tools/admin/cf-vault/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  version = "0.0.11";
+
+  src = fetchFromGitHub {
+    owner = "jacobbednarz";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-Imd9qeT4xg5ujVPLHSSqoteSPl9t97q3Oc4C/vzHphg=";
+  };
+
+  proxyVendor = true;
```
Comment (zowoq) [inline pkgs/tools/admin/cf-vault/default.nix:13]: `proxyVendor` is a no op with `vendorSha256 = null`
```suggestion
```

File: pkgs/tools/admin/cf-vault/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  vendorSha256 = null;
+
+  meta = with lib; {
+    description = ''
+      A tool for managing your Cloudflare credentials, securely..
+    '';
+    license = licenses.mit;
+    maintainers = with maintainers; [ viraptor ];
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (zowoq) [inline pkgs/tools/admin/cf-vault/default.nix:23]: Default for `buildGoModule`.
```suggestion
```

**Review by zowoq — CHANGES_REQUESTED** [review]

File: pkgs/tools/admin/cf-vault/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchFromGitHub {
+    owner = "jacobbednarz";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-Imd9qeT4xg5ujVPLHSSqoteSPl9t97q3Oc4C/vzHphg=";
+  };
+
+
+  vendorSha256 = null;
```
Comment (zowoq) [inline pkgs/tools/admin/cf-vault/default.nix:14]: Fails to build. Upstream doesn't have a vendor dir so this needs to be set.

```suggestion
  vendorSha256 = "sha256-PkmbVg5HnsUaSL/Kp3YJVdyzpjgvr/p9mKNmOubwXQA=";
```

File: pkgs/tools/admin/cf-vault/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  meta = with lib; {
+    description = ''
+      A tool for managing your Cloudflare credentials, securely..
+    '';
+    license = licenses.mit;
```
Comment (zowoq) [inline pkgs/tools/admin/cf-vault/default.nix:20]: Should have a `homepage`.

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -12692,6 +12692,12 @@
     githubId = 118959;
     name = "VinyMeuh";
   };
+  viraptor = {
```
Comment (zowoq) [inline maintainers/maintainer-list.nix:12695]: This commit can be dropped now.

**Pull request merged**